client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -886,10 +886,7 @@ Params withDetailed(boolean detailed) {
         }
 
         Params withWaitForCompletion(boolean waitForCompletion) {
-            if (waitForCompletion) {
-                return putParam("wait_for_completion", Boolean.TRUE.toString());
-            }
-            return this;
+            return putParam("wait_for_completion", Boolean.toString(waitForCompletion));
```

Comment:
that need to be to set false and pass through to the server.

Reply:
Ahh, this is due to the fact that `wait_for_completion` defaults to true in this API, right?

Reply:
exactly. See RestIndexUpgradeAction:60 
   `     if (request.paramAsBoolean("wait_for_completion", true)) { `